php-checks/src/main/java/org/sonar/php/checks/MissingMethodVisibilityCheck.java

```diff
@@ -20,81 +20,79 @@
 package org.sonar.php.checks;
 
 import com.google.common.collect.ImmutableSet;
-import com.sonar.sslr.api.AstNode;
 import org.sonar.api.server.rule.RulesDefinition;
 import org.sonar.check.Priority;
 import org.sonar.check.Rule;
 import org.sonar.php.api.PHPKeyword;
-import org.sonar.php.parser.PHPGrammar;
+import org.sonar.plugins.php.api.tree.Tree.Kind;
+import org.sonar.plugins.php.api.tree.declaration.ClassDeclarationTree;
+import org.sonar.plugins.php.api.tree.declaration.ClassMemberTree;
+import org.sonar.plugins.php.api.tree.declaration.MethodDeclarationTree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
 import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
 import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
-import org.sonar.squidbridge.checks.SquidCheck;
-import org.sonar.sslr.parser.LexerlessGrammar;
 
 import java.util.Set;
 
 @Rule(
-  key = "S1784",
+  key = MissingMethodVisibilityCheck.KEY,
   name = "Method visibility should be explicitly declared",
   priority = Priority.MINOR,
   tags = {Tags.CONVENTION, Tags.PSR2})
 @SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.UNDERSTANDABILITY)
 @SqaleConstantRemediation("2min")
-public class MissingMethodVisibilityCheck extends SquidCheck<LexerlessGrammar> {
+public class MissingMethodVisibilityCheck extends PHPVisitorCheck {
 
-  private static final Set<PHPKeyword> VISIBILITIES = ImmutableSet.of(
-    PHPKeyword.PRIVATE,
-    PHPKeyword.PROTECTED,
-    PHPKeyword.PUBLIC);
+  public static final String KEY = "S1784";
+  private static final String MESSAGE = "Explicitly mention the visibility of this %s \"%s\".";
+
+  private static final Set<String> VISIBILITIES = ImmutableSet.of(
+    PHPKeyword.PRIVATE.getValue(),
+    PHPKeyword.PROTECTED.getValue(),
+    PHPKeyword.PUBLIC.getValue());
 
   @Override
-  public void init() {
-    subscribeTo(
-      PHPGrammar.METHOD_DECLARATION);
+  public void visitClassDeclaration(ClassDeclarationTree tree) {
+    super.visitClassDeclaration(tree);
+
+    if (tree.is(Kind.CLASS_DECLARATION)) {
+
+      for (ClassMemberTree member : tree.members()) {
+
+        if (member.is(Kind.METHOD_DECLARATION)) {
+          checkMethod((MethodDeclarationTree) member, tree.name().text());
+        }
+      }
+    }
   }
 
-  @Override
-  public void visitNode(AstNode astNode) {
-    if (!hasVisibility(astNode)) {
-      getContext().createLineViolation(this, "Explicitly mention the visibility of this {0}.", astNode, getMessageForMethodName(astNode));
+  // fixme (Lena) : could be refactored with https://github.com/pynicolas/sonar-php/commit/c8ba74d43c0816871e928d9415da68791fbde5e8
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)